pkg/serializer/serializer.go

```diff
@@ -357,7 +357,7 @@ func (s *Serializer) SendIterableSeries(serieSource metrics.SerieSource) error {
 	failoverActiveForMRF, allowlistForMRF := s.getFailoverAllowlist()
 	failoverActiveForAutoscaling, allowlistForAutoscaling := s.getAutoscalingFailoverMetrics()
 	failoverActive := (failoverActiveForMRF && len(allowlistForMRF) > 0) || (failoverActiveForAutoscaling && len(allowlistForAutoscaling) > 0)
-	pipelines := make([]metricsserializer.Pipeline, 0, 3)
+	pipelines := make([]metricsserializer.Pipeline, 0, 4)
```

Comment:
We can probably leave counting to Go here:

```suggestion
	pipelines := []metricsserializer.Pipeline{}
```